src/rrules.jl

```diff
@@ -32,16 +32,12 @@ LazyPropagation(F::judiPropagator, q) = LazyPropagation(identity, F, q)
 
 for op in [:+, :-, :*, :/]
     @eval begin
-        $(op)(F::LazyPropagation, y::AbstractArray{T}) where T = $(op)(eval_prop(F), y)
-        $(op)(y::AbstractArray{T}, F::LazyPropagation) where T = $(op)(y, eval_prop(F))
+        $(op)(F::LazyPropagation, y::Union{AbstractArray{T}, T}) where T = $(op)(eval_prop(F), y)
```

Comment:
You are generalizing in a bad way. Unions aren't really nice for different abstract type. So Union of like Vector and Matrix is fine but for Scalar and AbstractArray it's not great.

You are also completely ignoring the setup here that make type separate through eval, that Union completely defeat the purpose.

Finally, you are again not considering what this type is about nor what you are trying to do and put here first way "that works" you could make up. The main point is to avoid evaluating PDEs when not necessary and you are not forcing potentially un-neccessary PDEs. a LazyPropagation is Linear, if you do `a .* Lazy` then it's the same as `Lazy(L.F, a .* L.q)` which does not evaluate anything.